src/panels/lovelace/cards/hui-entities-card.ts

```diff
@@ -218,7 +229,40 @@ class HuiEntitiesCard extends LitElement implements LovelaceCard {
               </h1>
             `}
         <div id="states" class="card-content">
-          ${this._configEntities!.map((entityConf) =>
+          ${(this._sortByState
+            ? this._configEntities!.sort((confA, confB) => {
+                const reverseSortOrder = this._reverseSortOrder ? 1 : -1;
+                const aBeforeB = -1 * reverseSortOrder;
+                const bBeforeA = -aBeforeB;
+
+                if (
```

Comment:
Should you return 0 if both A and B are invalid? Other sort algorithms in the frontend I've seen do this. I think this prevents jumbling up the entities in a random order if there are multiple invalid entities.

Reply:
Good point, I'm going to do it like you said